test/e2e/network/endpointslice.go

```diff
@@ -785,3 +792,56 @@ func createServiceReportErr(cs clientset.Interface, ns string, service *v1.Servi
 	framework.ExpectNoError(err, "error deleting Service")
 	return svc
 }
+
+// endpointSlicesEqual compare if the Endpoint and the EndpointSliceList contains the same endpoints values
+// as in addresses and ports, considering Ready and Unready addresses
+func endpointSlicesEqual(endpoints *v1.Endpoints, endpointSliceList *discoveryv1.EndpointSliceList) bool {
+	// get the apiserver endpoint addresses
+	epAddresses := sets.NewString()
+	epPorts := sets.NewInt32()
+	for _, subset := range endpoints.Subsets {
+		for _, addr := range subset.Addresses {
+			epAddresses.Insert(addr.IP)
+		}
+		for _, addr := range subset.NotReadyAddresses {
+			epAddresses.Insert(addr.IP)
+		}
+		for _, port := range subset.Ports {
+			epPorts.Insert(port.Port)
+		}
+	}
+	framework.Logf("Endpoints addresses: %v , ports: %v", epAddresses.List(), epPorts.List())
+
+	// Endpoints are single stack, and must match the primary IP family of the Service kubernetes.default
+	// However, EndpointSlices can be IPv4 or IPv6, we can only compare the Slices that match the same IP family
+	// framework.TestContext.ClusterIsIPv6() reports the IP family of the kubernetes.default service
+	isIPv6 := framework.TestContext.ClusterIsIPv6()
```

Comment:
Tiny nit: You can replace/dedup the ifs in the loop if you add an if here. Something like this here:
```suggestion
	var addrType discoveryv1.AddressType
	if framework.TestContext.ClusterIsIPv6() {
	  addrType = discoveryv1.AddressTypeIPv6
	} else {
	  addrType = discoveryv1.AddressTypeIPv4
	}
```

Followed by this in the loop:
```go
		if slice.AddressType != addrType {
			framework.Logf("Skipping slice %s: wanted %s family, got %s", slice.Name, addrType, slice.AddressType)
			continue
		}
```

Reply:
much cleaner, thanks